contracts/RelayerHub.sol

```diff
@@ -8,88 +8,192 @@ import "./System.sol";
-    }
-    emit paramChange(key, value);
-  }
-
-  function isRelayer(address sender) external override view returns (bool) {
-    return relayersExistMap[sender];
-  }
+contract RelayerHub is IRelayerHub, System, IParamSubscriber {
+    using SafeMath for uint256;
+
+    uint256 public constant INIT_REQUIRED_DEPOSIT = 1e20;
+    uint256 public constant INIT_DUES = 1e17;
+    address public constant WHITELIST_1 = 0xb005741528b86F5952469d80A8614591E3c5B632;
+    address public constant WHITELIST_2 = 0x446AA6E0DC65690403dF3F127750da1322941F3e;
+
+    uint256 public requiredDeposit; // have to keep it to not break the storage layout
+    uint256 public dues;
+
+    mapping(address => relayer) relayers; // old map holding the relayers which are to be allowed safe exit
+    mapping(address => bool) relayersExistMap;
+
+    struct relayer {
+        uint256 deposit;
+        uint256 dues;
+    }
+
+    struct manager {
+        uint256 dues;
+        bool registered;
+    }
+
+    mapping(address => manager) managers;
+    mapping(address => bool) relayManagersExistMap;
+    mapping(address => address) managerToRelayer;
+    mapping(address => bool) currentRelayers;
+
+    bool public whitelistInitDone;
+
+    modifier onlyNonRegisteredManager() {
+        require(relayManagersExistMap[msg.sender], "manager does not exist");
+        require(!managers[msg.sender].registered, "manager already registered");
+        _;
+    }
+
+    modifier onlyRegisteredManager() {
+        require(managers[msg.sender].registered, "manager not registered");
+        _;
+    }
+
+    modifier exist() {
+        require(relayersExistMap[msg.sender], "relayer do not exist");
+        _;
+    }
+
+    event relayerRegister(address _relayer);
+    event relayerUnRegister(address _relayer);
+    event paramChange(string key, bytes value);
+
+    event removeManagerEvent(address _removedManager);
+    event addManagerByGovEvent(address _addedManager);
+    event registerManagerEvent(address _registeredManager);
+    event updateRelayerEvent(address _from, address _to);
+
+    function init() external onlyNotInit {
+        requiredDeposit = INIT_REQUIRED_DEPOSIT;
+        dues = INIT_DUES;
+        alreadyInit = true;
+    }
+
+    function unregister() external exist onlyInit {
+        relayer memory r = relayers[msg.sender];
+        msg.sender.transfer(r.deposit.sub(r.dues));
+        address payable systemPayable = address(uint160(SYSTEM_REWARD_ADDR));
+        systemPayable.transfer(r.dues);
+        delete relayersExistMap[msg.sender];
+        delete relayers[msg.sender];
+        emit relayerUnRegister(msg.sender);
+    }
+
+    function whitelistInit() external {
+        require(!whitelistInitDone, "the whitelists already updated");
+        addInitRelayer(WHITELIST_1);
+        addInitRelayer(WHITELIST_2);
+        whitelistInitDone = true;
+    }
+
+    function addInitRelayer(address addr) internal {
+        managers[addr] = manager(dues, true);
```

Comment:
Why not do:
```
addManagerByGov(addr)
registerManagerAddRelayer(addr)
```
Much simple, and just need rename `addManagerByGov` to `addManager`

Reply:
It might require few additional changes if not an extra function since the `registerManagerAddRelayer()` function which will be removed and replaced by `updateRelayer()` and `updateRelayer()` can only be called by managers whereas with the current setup we are allowing anyone to call the `whitelistInit()` function to do the initialisation. 

So too much of a change to replace 3 lines of code. 😄